File: api/csharp/ToolGood.TextFilter.Api/Datas/Requests/TextReplaceRequest.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace ToolGood.TextFilter.Api.Datas.Requests
{
    public class TextReplaceRequest
    {
        /// <summary>
        /// 需要检测的文本
        /// </summary>
        public string Txt { get; set; }
        /// <summary>
        /// 是否跳过 Bidi 字符，默认 false
        /// </summary>
        public bool SkipBidi { get; set; }
        /// <summary>
        /// 替换符号, 默认 星号
        /// </summary>
        public char ReplaceChar { get; set; }
        /// <summary>
        /// 人工审核替换，默认true
        /// </summary>
        public bool ReviewReplace { get; set; }
        /// <summary>
        /// 联系字符串替换，默认true
        /// </summary>
        public bool ContactReplace { get; set; }
        /// <summary>
        /// 只显示位置，不显示匹配文字
        /// </summary>
        public bool OnlyPosition { get; set; }

        public TextReplaceRequest()
        {
            ReviewReplace = true;
            ContactReplace = true;
            SkipBidi = false;
            ReplaceChar = '*';
        }
        /// <summary>
        /// 
        /// </summary>
        /// <param name="txt">需要检测的文本</param>
        /// <param name="replaceChar">替换符号, 默认 星号</param>
        /// <param name="reviewReplace">人工审核替换，默认true</param>
        /// <param name="contactReplace">联系字符串替换，默认true</param>
        /// <param name="skipBidi">是否跳过 Bidi 字符</param>
        /// <param name="onlyPosition">只显示位置，不显示匹配文字</param>
        public TextReplaceRequest(string txt, char replaceChar = '*', bool reviewReplace = true, bool contactReplace = true, bool skipBidi = false, bool onlyPosition = false)
        {
            Txt = txt;
            SkipBidi = skipBidi;
            ReplaceChar = replaceChar;
            ReviewReplace = reviewReplace;
            ContactReplace = contactReplace;
            OnlyPosition = onlyPosition;
        }


#if GRPC
        public TextReplaceGrpcRequest ToGrpcRequest()
        {
            return new TextReplaceGrpcRequest() {
                Txt = this.Txt,
                SkipBidi = this.SkipBidi,
                ReplaceChar = this.ReplaceChar,
                ReviewReplace = this.ReviewReplace,
                ContactReplace = this.ContactReplace,
                OnlyPosition = OnlyPosition,

            };
        }
#endif


    }


}

File: api/java/toolgood.textfilter.api/src/main/java/toolgood/textfilter/api/Datas/Requests/TextReplaceRequest.java
package toolgood.textfilter.api.Datas.Requests;

import toolgood.textfilter.api.GrpcBase.TextFilter.TextReplaceGrpcRequest;

public class TextReplaceRequest {
    /**
     * 需要检测的文本
     */
    public String Txt;

    /**
     * 是否跳过 Bidi 字符，默认 false
     */
    public boolean SkipBidi;
    /**
     * 替换符号, 默认 星号
     */
    public char ReplaceChar;
    /**
     * 人工审核替换，默认true
     */
    public boolean ReviewReplace;
    /**
     * 联系字符串替换，默认true
     */
    public boolean ContactReplace;
    /**
     * 只显示位置，不显示匹配文字
     */
    public boolean OnlyPosition;

    public TextReplaceRequest() {
        ReviewReplace = true;
        ContactReplace = true;
        SkipBidi = false;
        ReplaceChar = '*';
    }

    /**
     * 
     * @param txt            需要检测的文本
     * @param replaceChar    替换符号, 默认 星号
     * @param reviewReplace  人工审核替换，默认true
     * @param contactReplace 联系字符串替换，默认true
     * @param skipBidi       是否跳过 Bidi 字符
     * @param onlyPosition   只显示位置，不显示匹配文字
     */
    public TextReplaceRequest(String txt, char replaceChar, boolean reviewReplace, boolean contactReplace,
            boolean skipBidi, boolean onlyPosition) {
        Txt = txt;
        SkipBidi = skipBidi;
        ReplaceChar = replaceChar;
        ReviewReplace = reviewReplace;
        ContactReplace = contactReplace;
        OnlyPosition = onlyPosition;
    }

    public TextReplaceGrpcRequest ToGrpcRequest() {
        return TextReplaceGrpcRequest.newBuilder().setTxt(Txt).setSkipBidi(SkipBidi).setReviewReplace(ReviewReplace)
                .setReplaceChar(ReplaceChar).setContactReplace(ContactReplace).setOnlyPosition(OnlyPosition).build();

    }

}
